src/libs/fileDownload/index.desktop.ts

```diff
@@ -0,0 +1,32 @@
+import ELECTRON_EVENTS from '@desktop/ELECTRON_EVENTS';
+import type {Options} from '@desktop/electronDownloadManagerType';
+import type {FileDownload} from './types';
+
+/**
+ * The function downloads an attachment on desktop platforms.
+ */
+const fileDownload: FileDownload = (url, fileName) => {
+    const options: Options = {
+        filename: fileName,
+        saveAs: true,
+    };
+    window.electron.send(ELECTRON_EVENTS.DOWNLOAD, {url, options});
+
+    return new Promise((resolve) => {
+        window.electron.on(ELECTRON_EVENTS.DOWNLOAD_STARTED, (...args: unknown[]) => {
+            if (!Array.isArray(args) || args[0] === null || typeof args[0] !== 'object' || !('url' in args[0])) {
+                return;
+            }
+
+            const {url: eventUrl} = args[0] as {url: string};
+
+            if (eventUrl !== url) {
+                return;
+            }
```

Comment:
Hm... Wouldn't the app be in a state of `eternal sunshine` if the any of the above condition becomes `true`? I mean, you might want to reject or resolve in here or the promise would go on forever.

Reply:
@allroundexperts intentionally done, further optimized with an additional comment
```js
window.electron.on(ELECTRON_EVENTS.DOWNLOAD_STARTED, (...args: unknown[]) => {
            const arg = Array.isArray(args) ? args[0] : null;
            const eventUrl = arg && typeof arg === 'object' && 'url' in arg ? arg.url : null;

            // This event is triggered for all active download instances. We intentionally keep other promises waiting.
            // Early resolution or rejection of other promises could prematurely stop the loading spinner or prevent the promise from being resolved.
            if (eventUrl === url) {
                resolve();
            }
        });
```